powersimdata/input/input_data.py

```diff
@@ -3,139 +3,46 @@
         file_name = scenario_info["id"] + "_" + field_name + "." + ext
-        return file_name, server_setup.INPUT_DIR
-
-
-def _check_field(field_name):
-    """Checks field name.
-
-    :param str field_name: *'demand'*, *'hydro'*, *'solar'*, *'wind'*,
-        *'demand_flexibility_up'*, *'demand_flexibility_dn'*,
-        *'demand_flexibility_cost_up'*, *'demand_flexibility_cost_dn'*, *'ct'*, or
-        *'grid'*.
-    :raises ValueError: if not *'demand'*, *'hydro'*, *'solar'*, *'wind'*,
-        *'demand_flexibility_up'*, *'demand_flexibility_dn'*,
-        *'demand_flexibility_cost_up'*, *'demand_flexibility_cost_dn'*, *'ct'*, or
-        *'grid'*.
-    """
-    possible = list(_file_extension.keys())
-    if field_name not in possible:
-        raise ValueError("Only %s data can be loaded" % " | ".join(possible))
-
-
-def _read(f, filepath):
-    """Read data from file object
-
-    :param io.IOBase f: a file handle
-    :param str filepath: the filepath corresponding to f
-    :raises ValueError: if extension is unknown.
-    :return: object -- the result
-    """
-    ext = os.path.basename(filepath).split(".")[-1]
-    if ext == "pkl":
-        data = pd.read_pickle(f)
-    elif ext == "csv":
-        data = pd.read_csv(f, index_col=0, parse_dates=True)
-        if "demand_flexibility" in filepath:
-            data.columns = data.columns.astype(str)
-        else:
-            data.columns = data.columns.astype(int)
-    elif ext == "mat":
-        # get fully qualified local path to matfile
-        data = os.path.abspath(filepath)
-    else:
-        raise ValueError("Unknown extension! %s" % ext)
-
-    return data
+        return "/".join([*server_setup.INPUT_DIR, file_name])
 
+    def _read(self, f, path):
+        """Read data from file object
 
-class InputData:
-    """Load input data."""
-
-    def __init__(self):
-        """Constructor."""
-        self.data_access = Context.get_data_access()
-
-    def get_data(self, scenario_info, field_name):
-        """Returns data either from server or local directory.
-
-        :param dict scenario_info: scenario information.
-        :param str field_name: *'demand'*, *'hydro'*, *'solar'*, *'wind'*,
-            *'demand_flexibility_up'*, *'demand_flexibility_dn'*,
-            *'demand_flexibility_cost_up'*, *'demand_flexibility_cost_dn'*, *'ct'*, or
-            *'grid'*.
-        :return: (*pandas.DataFrame*, *dict*, or *str*) -- demand, hydro, solar, wind,
-            demand_flexibility_up, demand_flexibility_dn, demand_flexibility_cost_up,
-            or demand_flexibility_cost_dn as a data frame; change table as a dictionary;
-            or the path to a matfile enclosing the grid data.
-        :raises FileNotFoundError: if file not found on local machine.
+        :param io.IOBase f: an open file object
+        :param str path: the path corresponding to f
+        :raises ValueError: if extension is unknown.
+        :return: object -- the result
```

Comment:
:return: (*object*). Also, we could be more explicit on what the return object is. Could be either a `dict` (change table) or a `Grid` object